### PR TITLE
refactor() BREAKING: align line to other classes for positioning

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

Line has some peculiar code that could be removed in favor of standard methods we use everyewhere else.

## Description

Trying to remove the custom code for the standard setPostionByOrigin and makeBoundingBoxFromPoints, leaving open the issue with changes in positioning.

This won't affect old saved data but affect lines that are created without left,top and just points.
I would seem normal that if i have a line that extends from `5,15 to 15,5` it's center is in `10, 10` regardless of strokeWidth and that the left, top and originX or originY are assigned depending from the options or the defaults but not from the points themselves

Today you get different originX and originY depending on the order of your points, like if x1 or y1 are the origin, and unless you specified a different origin.

Make sense to remove those contraddictions but there are cases in which the lines aren't anymore in the previous places when initialized with the same points.

### CODE REVIEW

**Comment by ShaMan123** [comment]
I know you don't want more changes...
However, why not to completely remove this or make a subclass of Polyline, overriding `toObject/fromObject/fromElement`?

**Comment by asturur** [comment]
Because you can already have a polyline with 2 points.
I don't like this class but i don't want to risk svg import now.
We would need to keep everything and what would go away is basically the render code for the line and nothing more.

**Comment by asturur** [comment]
Ideally both polyon, polyline and line just extend Path limiting some functionality, but we need a smart way to handle the points handling afterwards.

**Comment by asturur** [comment]
I m amazed that the more es6 i use, the more the bundle grow, i think is all safari fault.

**Comment by ShaMan123** [comment]
It is really strange the stats from the last prs.
I don't pay enough attention to that.
Just see readable es6 code and smile...

**Comment by asturur** [comment]
is just because every time we add a deconstruct in the arguments safari freaks out.
Is ok one day we can move to safari 13 and the bundle will shrink.

**Comment by asturur** [comment]
you should see the template with parameters what does it do:
```js
    return ['<line ', 'COMMON_PARTS', "x1=\"".concat(x1, "\" y1=\"").concat(y1, "\" x2=\"").concat(x2, "\" y2=\"").concat(y2, "\" />\n")];

```

**Comment by ShaMan123** [comment]
Web dev can just go smoothly even when things start to workout

**Comment by ShaMan123** [comment]
> Web dev can just go smoothly even when things start to workout

Can't

File: src/shapes/Line.ts [hunk 1]
```diff
@@ -70,24 +71,28 @@ export class Line<
...
-    this.left = left ?? this._getLeftToOriginX();
-    this.top = top ?? this._getTopToOriginY();
+  _setWidthHeight() {
+    const { x1, y1, x2, y2 } = this;
+    this.width = Math.abs(x2 - x1);
+    this.height = Math.abs(y2 - y1);
+    const { left, top, width, height } = makeBoundingBoxFromPoints([
+      { x: x1, y: y1 },
+      { x: x2, y: y2 },
+    ]);
```
Comment (ShaMan123) [inline src/shapes/Line.ts:93]: redundant calc of width/height
obtain from `makeBoundingBoxFromPoints`

**Pull request merged**